GVFS/GVFS.Common/Git/GitSsl.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.Common.X509Certificates;
+
+namespace GVFS.Common.Git
+{
+    public class GitSsl
+    {
+        private readonly string certificatePathOrSubjectCommonName;
+        private readonly bool isCertificatePasswordProtected;
+        private readonly Func<SystemCertificateStore> createCertificateStore;
+        private readonly CertificateVerifier certificateVerifier;
+        private readonly PhysicalFileSystem fileSystem;
+
+        public GitSsl(
+            IDictionary<string, GitConfigSetting> configSettings,
+            Func<SystemCertificateStore> createCertificateStore = null,
+            CertificateVerifier certificateVerifier = null,
+            PhysicalFileSystem fileSystem = null) : this(createCertificateStore, certificateVerifier, fileSystem)
+        {
+            if (configSettings != null)
+            {
+                if (configSettings.TryGetValue(GitConfigSetting.HttpSslCert, out GitConfigSetting sslCerts))
+                {
+                    this.certificatePathOrSubjectCommonName = sslCerts.Values.Last();
+                }
+
+                this.isCertificatePasswordProtected = SetBoolSettingOrThrow(configSettings, GitConfigSetting.HttpSslCertPasswordProtected, this.isCertificatePasswordProtected);
+
+                this.ShouldVerify = SetBoolSettingOrThrow(configSettings, GitConfigSetting.HttpSslVerify, this.ShouldVerify);
+            }
+        }
+
+        private GitSsl(Func<SystemCertificateStore> certificateStoreFactory, CertificateVerifier certificateVerifier, PhysicalFileSystem fileSystem)

```

Comment:
Parameter is `certificateStoreFactory` and `createCertificateStore` above.  They should probably match.